pkg/utils/apiutil/serverapi/middleware.go

```diff
@@ -200,11 +202,8 @@ func (h *redirector) ServeHTTP(w http.ResponseWriter, r *http.Request, next http
 			return
 		}
 		clientUrls = append(clientUrls, targetAddr)
-		// Add a header to the response, this is not a failure injection
-		// it is used for testing, to check whether the request is forwarded to the micro service
-		failpoint.Inject("checkHeader", func() {
-			w.Header().Set(apiutil.ForwardToMicroServiceHeader, "true")
-		})
+		// Add a header to the response, it is used to mark whether the request has been forwarded to the micro service.
+		w.Header().Add(apiutil.XForwardedToMicroServiceHeader, "true")
```

Comment:
Or can we can keep failpoint?  If `len(r.Header.Get(apiutil.XForbiddenForwardToMicroServiceHeader)) > 0` or failpoint enabled, we add the header into resp.

Reply:
If `len(r.Header.Get(apiutil.XForbiddenForwardToMicroServiceHeader)) > 0` , code will not reach here

Reply:
If it is as you say, then the new check in `pdctl/global.go` is also unreachable

Reply:
But if we use curl or pdctl to call api, how do we tell if it's coming from the scheduling server or the api server?